ipld/src/lib.rs

```diff
@@ -14,17 +20,84 @@ use serde::de::DeserializeOwned;
 use serde::Serialize;
 use std::collections::BTreeMap;
 
-/// Represents IPLD data structure used when serializing and deserializing data
+/// Represents IPLD data structure used when serializing and deserializing data.
 #[derive(Debug, Clone, PartialEq)]
 pub enum Ipld {
+    /// Represents a null value.
+    ///
+    /// ```
+    /// # use forest_ipld::ipld;
+    /// let v = ipld!(null);
```

Comment:
Feels weird having a doc test for how to use an Enum... Just adds to test compilation and running without much value. Your documentation of the macro is pretty good already

Reply:
Made more sense to me to document how to construct each variant instead of doing all on the Ipld enum. These are pretty lightweight but I can add `no_run` to them or remove, just nice to have examples for each variant for practical use imo.